examples/summarization/bart/run_bart_sum.py

```diff
@@ -0,0 +1,247 @@
+import argparse
+import glob
+import logging
+import os
+import random
+
+import numpy as np
+import pytorch_lightning as pl
+import torch
+import torch.nn.functional as F
+from torch.utils.data import DataLoader
+
+from transformers import AdamW, BartTokenizer, get_linear_schedule_with_warmup
+from transformers.modeling_bart import BartForConditionalGeneration
+from utils import CnnDailyMailDataset, add_generic_args
+
+
+logger = logging.getLogger(__name__)
+
+
+class BartSystem(pl.LightningModule):
+    def __init__(self, hparams):
+        super(BartSystem, self).__init__()
+        self.hparams = hparams
+        self.bart = BartForConditionalGeneration.from_pretrained("bart-large", output_past=True)
+
+        self.tokenizer = BartTokenizer.from_pretrained("bart-large")
+        self.criterion = torch.nn.CrossEntropyLoss()
+
+    def forward(self, input_ids, attention_mask=None, decoder_input_ids=None, decoder_attention_mask=None):
+        return self.bart(
+            input_ids,
+            attention_mask=attention_mask,
+            decoder_input_ids=decoder_input_ids,
+            decoder_attention_mask=decoder_attention_mask,
+        )
+
+    def _step(self, batch):
+        outputs = self.forward(
+            batch["source_ids"], attention_mask=batch["source_mask"], decoder_input_ids=batch["target_ids"]
+        )
+
+        out = outputs[0]
+
+        logits = F.log_softmax(out, dim=-1)
+        y = batch["target_ids"]
+        norm = (y != self.tokenizer.pad_token_id).data.sum()
+
+        targets = y.clone()
+        targets[y == self.tokenizer.pad_token_id] = -100
+        loss = self.criterion(logits.contiguous().view(-1, logits.size(-1)), targets.contiguous().view(-1)) / norm
+        return loss
```

Comment:
Can we use the model's built in functionality for handling the loss (when you pass kwarg `lm_labels`), or no? I'm not familiar with the `BartForConditionalGeneration` class. 

From the [docstring](https://github.com/huggingface/transformers/blob/47591763137f17021928e686ef171f25c240f076/src/transformers/modeling_bart.py#L1053-L1059):
```
        tokenizer = BartTokenizer.from_pretrained('bart-large')
        model = BartForSequenceClassification.from_pretrained('bart-large')
        input_ids = torch.tensor(tokenizer.encode("Hello, my dog is cute",
        add_special_tokens=True)).unsqueeze(0)  # Batch size 1
        labels = torch.tensor([1]).unsqueeze(0)  # Batch size 1
        outputs = model(input_ids, labels=labels)
        loss, logits = outputs[:2]
```